bot/src/keeper/surplus.ts

```diff
@@ -0,0 +1,109 @@
+import { SurplusAuctionActive } from 'auctions-core/src/types';
+import { bidToSurplusAuction, enrichSurplusAuction } from 'auctions-core/src/surplus';
+import getSigner from 'auctions-core/src/signer';
+import { fetchAllowanceAmountMKR, setAllowanceAmountMKR } from 'auctions-core/src/authorizations';
+import { KEEPER_MINIMUM_NET_PROFIT_DAI } from '../variables';
+import { isSetupCompleted } from './setup';
+import BigNumber from '~/../core/src/bignumber';
+
+const currentlyExecutedAuctions = new Set();
+
+const checkAndParticipateIfPossible = async function (network: string, auction: SurplusAuctionActive) {
+    // check if setupKeeper hasn't run
+    if (!isSetupCompleted) {
+        return;
+    }
+
+    const signer = await getSigner(network);
+
+    // enrich the auction with more numbers
+    const auctionTransaction = await enrichSurplusAuction(network, auction);
+
+    // check if auction became inactive or finished
+    if (auctionTransaction.state === 'ready-for-collection') {
```

Comment:
Do we want to check if the user is the winner and then collect the auction in this case?